torch/_functorch/vmap.py

```diff
@@ -391,19 +398,114 @@ def vmap(
         sequences out of the box.
     """
     _check_randomness_arg(randomness)
+    if not (chunk_size is None or chunk_size > 0):
+        raise ValueError("vmap: `chunk_size` should be greater than 0.")
+
+    def _get_chunked_inputs(flat_args, flat_in_dims, batch_size, chunk_size):
+        split_idxs = (batch_size,)
+        if chunk_size is not None and chunk_size > 0:
+            n_chunks = n_chunks = batch_size // chunk_size
+            chunk_numels = [chunk_size] * n_chunks
+            # remainder chunk
+            chunk_numels.append(batch_size % chunk_size)
+            split_idxs = tuple(itertools.accumulate(chunk_numels))
```

Comment:
Can we share the chunk size computation code with that of jacrev? That would make it clearer that the behavior is the same